intro/sdk_code_guidelines.md

```diff
@@ -0,0 +1,159 @@
+  ``` 
+* Remove the [Patching DSL](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/README.md#patching-dsl) (`patchPluginXml{}`) section.
+  It is not needed in SDK samples.
+* Add the `javadoc` task to build documentation for the code sample:
+  ```groovy
+    // Force javadoc rebuild before jar is built
+    jar.dependsOn javadoc
+  ```
+
+## plugin.xml Conventions
+A consistent structure for the [`plugin.xml`](/basics/plugin_structure/plugin_configuration_file.md) configuration file of an SDK code sample is important because we want to draw attention to the unique parts of the file for a plugin.
+Inconsistent element order is visually noisy.
+Comment profusely about unique elements and configurations, and comment sparingly for the rest.
+
+The sequence of elements in an SDK code sample `plugin.xml` file is:
+* `<id>` Use the fully qualified [Plugin ID](#plugin-id-conventions).
+* `<name>` The name value does not have to match the [Plugin Name](#naming-conventions-for-sdk-plugins).
+  It might reflect the functionality of the plugin.
+  The name must start with "SDK:".
+* `<version>` The code sample's version in MAJOR.MINOR.FIX format.
+  * MAJOR corresponds to a significant upgrade in functionality.
+  * MINOR corresponds to minor refactoring and small improvements in functionality. 
+  * FIX corresponds to changes that fix problems without significant refactoring.
+* `<idea-version/>` Set the attributes:
+  * `since-build` attribute to the earliest compatible build number of the IntelliJ Platform.
+  * `until-build` attribute to an empty string `""`.
+    SDK code samples are reviewed with every major release to ensure compatibility with the latest IntelliJ Platform.
+* `<description>` is a succinct explanation of what is being demonstrated and how a user would access the functionality.
+* `<change-notes>` is an ordered list by version numbers with a brief description of changes for each version.
+* `<vendor>` Set the value to `IntelliJ Platform SDK`.
+  Set the attributes:
+  * `email` to a placeholder value `"faux-email@jetbrains.com"` 
+  * `url` to the JetBrains plugin repository `"https://plugins.jetbrains.com"`
+* Except for [special cases](/basics/plugin_structure/plugin_configuration_file.md) these elements are not used in SDK plugins.
+  * `<depends>`  
+  * `<helpset>` 
+  * `<resource-bundle>`
+* `<application-components>` Deprecated and should not be used for new plugins.
+* `<module-components>` Omit if not specifically used.
+* `<project-components>` Omit if not specifically used.
+* `<extensionPoints>` Generally not used by SDK plugins.
+* `<actions>` Omit if not specifically used.
+* `<extensions>` Omit if not specifically used.
+
+## Testing
+IntelliJ Platform SDK code samples should be built and tested against the `since-build` version.
+
+Code samples should build cleanly, with no warnings or errors, and new code samples should pass all IntelliJ IDEA code inspections.
```

Comment:
"inspections" as in "all by default enabled inspections" ?

Reply:
Yes, I will clarify.